### DISCUSSION TITLE
Work in progress responsive markdown images

### DISCUSSION THREAD

**Discussion by danielfdickinson** [adityatelange/hugo-PaperMod#562] [Ideas]
I thought you might be interested in the work I am doing on responsive images for markdown (plain and figure shortcode).

This is based on #514 and #544 and additional work.

https://gist.github.com/danielfdickinson/a71f7e8b7196ef9d35f38d626afbe134

**Comment by laywens** [comment]
I am really interested in this! This is the solution I've been trying to find myself but my attempts at writing the shortcodes are failing...

**Comment by laywens** [comment]
@danielfdickinson I'm trying your solution but I think I'm doing something wrong. The images are showing up "squeezed" into the page.

What I want to see is: 
![Screen Shot 2021-10-01 at 12 58 42 PM](https://user-images.githubusercontent.com/7918039/135679499-a405bdf4-12bc-42c0-babc-19572015dc39.png)

But what I am seeing instead is this:
![Screen Shot 2021-10-01 at 12 59 52 PM](https://user-images.githubusercontent.com/7918039/135679517-ed6c2970-4e2d-43f5-842a-992d984ec7cc.png)

I read the gist. 
1. I have `figure.html render-image.html responsive-images.html` all in their correct directories. 
2. I have set my params in `config.yml`   
```
responsiveImages: true 
convertImagesTo: "webp"
```
3. I have my images in the page bundle.

4. The shortcode I'm using in my markdown is:
`{{< figure src="test-CAD-vs-pointCloud.png" caption="The caption text" >}}`

5. I can tell that the responsive images are being created because the webpage html code shows this for the figure:
```
<figure>
  <img loading=lazy
  srcset="/posts/aligning-models-with-reality/test-CAD-vs-pointCloud_hu4d1f5c74b9c7222ac2730f25743ae623_1337237_8eb782837e231b2a035e1faac41d1569.webp
  360w,/posts/aligning-models-with-reality/test-CAD-vs-pointCloud_hu4d1f5c74b9c7222ac2730f25743ae623_1337237_b330c7daa8b02f93eb88c17241e099f7.webp
  480w,/posts/aligning-models-with-reality/test-CAD-vs-pointCloud_hu4d1f5c74b9c7222ac2730f25743ae623_1337237_9e66fe6e35b12b7102f29458cf9a2776.webp
  720w,/posts/aligning-models-with-reality/test-CAD-vs-pointCloud_hu4d1f5c74b9c7222ac2730f25743ae623_1337237_df41eb2e06e89adb616af7ad87b552e8.webp
  1080w,/posts/aligning-models-with-reality/test-CAD-vs-pointCloud_hu4d1f5c74b9c7222ac2730f25743ae623_1337237_2a4c10024d160e816a9750d8b68c7854.webp
  1500w,/posts/aligning-models-with-reality/test-CAD-vs-pointCloud_hu4d1f5c74b9c7222ac2730f25743ae623_1337237_c23916e8f9a77300222e69cbc18cc3b6.webp"
  sizes="(width: 360px) 360px, (width: 480px) 480px, (width: 768px) 720px,
  (width: 1200px) 1080px, 60vw"
  src=/posts/aligning-models-with-reality/test-CAD-vs-pointCloud_hu4d1f5c74b9c7222ac2730f25743ae623_1337237_c23916e8f9a77300222e69cbc18cc3b6.webp
  alt="The caption text" width=1920 height=612>
  <figcaption>
    <p>The caption text</p>
  </figcaption>
</figure>
```

So what do you think I'm doing wrong? How can I get the image to fit within the constraints of the browser window and render correctly?

**Comment by danielfdickinson** [comment]
@shanewasley Hi. Glad you are interested! You will need to use some CSS to adjust the height and width of the images. I would recommend setting/adding a class name to make it easier to apply the CSS.

HTH

If you need more details, please let me know, and I'll provide more when I get some time.

**Reply by laywens** [reply]
Ah! Thank you, I'll give it a shot.

**Comment by danielfdickinson** [comment]
FYI, I am incorporating this (and more) into a Hugo module https://github.com/danielfdickinson/hugo-dfd-responsive-images (not ready for release yet, but if you are interested in contributing, please drop by). I'm hoping to be able to use this module with this theme but that will be up the theme maintainer. I'll be using myself for certain, and I hope it will get picked up by others.

**Reply by danielfdickinson** [reply]
I've updated the [README](https://github.com/danielfdickinson/hugo-dfd-responsive-images#readme) and have a [Demo site](https://hugo-dfd-responsive-images.wildtechgarden.ca/) which also has a [a version of the README](https://hugo-dfd-responsive-images.wildtechgarden.ca/docs/readme/).

Please let me know if you need more info!

**Comment by pourmand1376** [comment]
Hi @danielfdickinson, The repo you mentioned is archived. Do you know how I can add this thing to my blog?

**Reply by danielfdickinson** [reply]
I'd check out https://discourse.gohugo.io/t/deprecating-my-modules-on-github/41127, or start a new topic on https://discourse.gohugio.io asking about available modules / repos for responsive images.

Otherwise, if you are familiar with Go or Hugo modules, you could clone the repo and create your own version (you'd need to update the module config to use your repo URL instead the archived one, obviously).
